examples/pic.cpp

```diff
@@ -0,0 +1,363 @@
+    infmt.set_input(husky::Context::get_param("input"));
+
+    int dimension = std::stoi(husky::Context::get_param("dimension"));
+
+    // 1. Create and globalize ndpoint objects
+    husky::ObjList<NDpoint> ndpoint_list;
+    auto parse_ndpoint = [&ndpoint_list](boost::string_ref& chunk) {
+        if (chunk.size() == 0)
+            return;
+        boost::char_separator<char> sep(" \t");
+        boost::tokenizer<boost::char_separator<char>> tok(chunk, sep);
+        boost::tokenizer<boost::char_separator<char>>::iterator it = tok.begin();
+        int id = stoi(*it++);
+        std::vector<float> coords;
+        while (it != tok.end()) {
+            coords.push_back(stod(*it++));
+        }
+        ndpoint_list.add_object(NDpoint(id, coords));
+    };
+    load(infmt, parse_ndpoint);
+    globalize(ndpoint_list);
+    husky::base::log_msg( "ndpoint_list size " + std::to_string( ndpoint_list.get_size()));
+
+    husky::base::log_msg( "part1 ok ");
+    
+    // 2. calculate the variance used in similarity function
+    husky::ObjList<DimStat> dimstat_list;
+    globalize(dimstat_list);
+
+    husky::lib::Aggregator<int> sumAgg;
+    auto& ac = husky::lib::AggregatorFactory::get_channel();
+
+    auto& meanCh =
+        husky::ChannelFactory::create_push_combined_channel<float, husky::SumCombiner<float>>(ndpoint_list, dimstat_list);
+    auto& varCh =
+        husky::ChannelFactory::create_push_combined_channel<float, husky::SumCombiner<float>>(ndpoint_list, dimstat_list);
+    list_execute(ndpoint_list, {}, {&ac, &meanCh, &varCh}, [&sumAgg, &meanCh, &varCh](NDpoint& p) {
+        int idx=0;
+        for (auto& aVal : p.coords) {
+            meanCh.push(      aVal, idx);
+            varCh .push( aVal*aVal, idx);
+            idx++;
+        }
+        sumAgg.update(1);
+    });
+
+    list_execute(dimstat_list, [&sumAgg, &meanCh, &varCh](DimStat& d) {
+        int count = sumAgg.get_value(); 
+        d.mean = meanCh.get(d)/count;
+        d.var  = varCh.get(d)/count - d.mean*d.mean; //Var(x) = <x^2> - <x><x>
+        husky::base::log_msg( "stat " + std::to_string(d.id()) + "  " + std::to_string(count) + " " + std::to_string(d.mean) + " " + std::to_string(d.var));
+    });
+
+    husky::base::log_msg( "part2 ok ");
+
+    // 3. calculate the affinity matrix
+    husky::ObjList<MatElem> affMatElem_list;
+    globalize(affMatElem_list);
+    auto& pairCh =
+        husky::ChannelFactory::create_push_channel<NDpoint>(ndpoint_list, affMatElem_list);
+
+    //FIXME:: seems stupid to create a Channel just to make a list accessable by all worker?
+    auto& dimstatBCh = husky::ChannelFactory::create_broadcast_channel<int,DimStat>(dimstat_list);
+    list_execute(dimstat_list, [&](DimStat& d) {
+        dimstatBCh.broadcast( d.id(), d);
+    });
+
+    ////Why this doesn't work? I get vec of size 0 even dimstat_list is globalized
+    //std::vector<DimStat> local_dimstat_list = dimstat_list.get_data(); 
```

Comment:
You may discuss with us but don't leave them in the code. By the way, you are copying a std::vector.